lib/smwc_bot/fridge.ex

```diff
@@ -0,0 +1,116 @@
+defmodule SMWCBot.Fridge do
+  @moduledoc """
+  Handles command cooldown
+  """
+
+  use GenServer
+
+  require Logger
+
+  @ttl_seconds 60
+
+
+  @spec start_link(keyword()) :: GenServer.on_start()
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+  end
+
+  def insert(map) when is_map(map) do
+    map_to_sha256(map)
+    |> insert()
+  end
+
+  def insert(hash) when is_bitstring(hash) do
+    GenServer.cast(__MODULE__, {:insert, %{:key => hash, :val => NaiveDateTime.utc_now() }})
+  end
+
+  def upsert(map) when is_map(map) do
+    map_to_sha256(map)
+    |> upsert()
+  end
+
+  def upsert(hash) when is_bitstring(hash) do
+    if is_expired?(hash) do
+      remove(hash)
+      insert(hash)
+    end
+  end
+
+  def is_expired?(map) when is_map(map) do
+    map_to_sha256(map)
+    |> is_expired?()
+  end
+
+  def is_expired?(hash) when is_bitstring(hash) do
+    case Map.has_key?(get_contents(), hash) do
+      true -> diff_from_element(hash) + @ttl_seconds <= 0
+      _ -> true
+    end
+  end
+
+  defp diff_from_element(map) when is_map(map) do
+    map_to_sha256(map)
+    |> diff_from_element()
+  end
+
+  defp diff_from_element(hash) when is_bitstring(hash) do
+    get_contents()
+    |> Map.get(hash, NaiveDateTime.utc_now())
+    |> NaiveDateTime.diff(NaiveDateTime.utc_now())
+  end
+
+  def get_contents() do
+    GenServer.call(__MODULE__, :contents)
+  end
+
+  def remove(map) when is_map(map) do
+    map_to_sha256(map)
+    |> remove()
+  end
+
+  def remove(hash) when is_bitstring(hash) do
+    if Map.has_key?(get_contents(), hash) do
+      GenServer.call(__MODULE__, {:delete, hash})
+    else
+      true
+    end
+  end
+
+  defp map_to_sha256(map) do
+    map
+    |> Map.to_list()
+    |> Enum.reduce("", fn {k,v}, acc -> "#{k}#{v}" <> acc end)
+    |> string_to_sha256()
```

Comment:
I think string-building is more efficient if you make the list and then just do `IO.iodata_to_binary()`